### PR TITLE
MonadReader instance for EitherT

### PR DESCRIPTION
**Pull request by kaychaks**

### CODE REVIEW

File: tests/src/test/scala/scalaz/EitherTTest.scala [hunk 1]
```diff
@@ -121,6 +121,8 @@ object EitherTTest extends SpecLite {
     def nondeterminismMonad[A, F[_] : Nondeterminism] = Monad[EitherT[A, F, ?]]
     def nondeterminismFunctor[A, F[_] : Nondeterminism: BindRec: Traverse] = Functor[EitherT[A, F, ?]]
     def nondeterminismMonad[A, F[_] : Nondeterminism: BindRec: Traverse] = Monad[EitherT[A, F, ?]]
+
+    def monadReader[E, R, F[_] : Monad](implicit F0: MonadReader[F, R]) = MonadReader[EitherT[E, F, ?], R]
```
Comment (xuwei-k) [inline tests/src/test/scala/scalaz/EitherTTest.scala]: please add here

```diff
     def bitraverse[F[_] : Traverse] = Bitraverse[EitherT[?, F, ?]]
+    def monadReader[E, R, F[_]](implicit F0: MonadReader[F, R]) = MonadReader[EitherT[E, F, ?], R]
 
     // checking absence of ambiguity
     def functor[A, F[_] : BindRec] = Functor[EitherT[A, F, ?]]
@@ -121,8 +122,6 @@ object EitherTTest extends SpecLite {
     def nondeterminismMonad[A, F[_] : Nondeterminism] = Monad[EitherT[A, F, ?]]
     def nondeterminismFunctor[A, F[_] : Nondeterminism: BindRec: Traverse] = Functor[EitherT[A, F, ?]]
     def nondeterminismMonad[A, F[_] : Nondeterminism: BindRec: Traverse] = Monad[EitherT[A, F, ?]]
-
-    def monadReader[E, R, F[_] : Monad](implicit F0: MonadReader[F, R]) = MonadReader[EitherT[E, F, ?], R]
   }
```

File: core/src/main/scala/scalaz/EitherT.scala [hunk 2]
```diff
@@ -330,6 +330,12 @@ sealed abstract class EitherTInstances2 extends EitherTInstances3 {
     new EitherTMonadError[F, E] {
       implicit def F = F0
     }
+
+  implicit def eitherTMonadReader[E, F[_], R](implicit F0: MonadReader[F, R], M0: Monad[F]): MonadReader[EitherT[E, F, ?], R] =
```
Comment (xuwei-k) [inline core/src/main/scala/scalaz/EitherT.scala]:
```
[error]  both method eitherTMonadReader in class EitherTInstances2 of type [E, F[_], R](implicit F0: scalaz.MonadReader[F,R], implicit M0: scalaz.Monad[F])scalaz.MonadReader[[γ$6$]scalaz.EitherT[E,F,γ$6$],R]
[error]  and method eitherTMonadError in class EitherTInstances2 of type [F[_], E](implicit F0: scalaz.Monad[F])scalaz.MonadError[[γ$5$]scalaz.EitherT[E,F,γ$5$],E]
[error]  match expected type scalaz.Monad[[γ$45$]scalaz.EitherT[E,F,γ$45$]]
[error]     def monadReader[E, R, F[_]](implicit F0: MonadReader[F, R]) = Monad[EitherT[E, F, ?]]
[error]                                                                        ^
```

`eitherTMonadReader` and `eitherTMonadError` ambiguous if add following test in EitherTTest.

```scala
def monadReader[E, R, F[_]](implicit F0: MonadReader[F, R]) = Monad[EitherT[E, F, ?]]
```

File: .gitignore [hunk 3]
```diff
@@ -14,3 +15,4 @@ tags
 .classpath
 .project
 .settings
+.bloop
```
Comment (xuwei-k) [inline .gitignore]: please split commits any refactoring or cosmetic changes 🙏

File: core/src/main/scala/scalaz/EitherT.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaychaks) [reply core/src/main/scala/scalaz/EitherT.scala]: Thanks, made the changes in a new commit

File: tests/src/test/scala/scalaz/EitherTTest.scala (hunk 1 above)
Comment (kaychaks) [reply tests/src/test/scala/scalaz/EitherTTest.scala]: Thanks. Done in the new commit

File: core/src/main/scala/scalaz/EitherT.scala (hunk 2 above)
Comment (kaychaks) [reply core/src/main/scala/scalaz/EitherT.scala]: Reorganised `eitherTMonadReader` instance to resolve the ambiguity error. Please check.

File: .gitignore (hunk 3 above)
Comment (kaychaks) [reply .gitignore]: Now it iss in its own commit. thanks for pointing out

**Pull request merged**